addons/web/static/src/views/fields/formatters.js

```diff
@@ -313,7 +313,7 @@ export function formatMonetary(value, options = {}) {
 
     let currencyId = options.currencyId;
     if (!currencyId && options.data) {
-        const currencyField = options.currencyField || "currency_id";
+        const currencyField = options.currencyField || options.field && options.field.currency_field || "currency_id";
```

Comment:
@h4818 Could you try 
```js
const currencyField = options.currencyId || "currency_id";
```
It seems it is what it is supposed to be looking at the `monetary_field`.

Also, we'll need a qunit test to go with that, prolly in monetary_field_test

Reply:
wait currencyId is taken and tested right above..
But basically, the monetary field takes the currencyField and and get the currencyId out of it, so you shouldn't have to get it back there

Reply:
maybe the `currencyId` getter inside the field code is not right for your specific use case

Reply:
@SimonGenin IIANM the monetary_field is not actually instantiated here - instead `list_renderer` is calling `getFormattedValue` directly - and `options.data["currency_id"]` is not the correct currency.
@epictete can you confirm that?

https://github.com/odoo/odoo/blob/33b44ce84ba0d00003d7c9b6500e60c1bb247672/addons/web/static/src/views/list/list_renderer.xml#L228

Reply:
Why not pass the currencyField in the formatOptions ? (getFormattedValue(column, record))

Reply:
```js
     getFormattedValue(column, record) {
        const fieldName = column.name;
        const field = this.fields[fieldName];
        const formatter = formatters.get(field.type, (val) => val);
        const formatOptions = {
            escape: false,
            data: record.data,
            isPassword: "password" in column.rawAttrs,
            digits: column.rawAttrs.digits ? JSON.parse(column.rawAttrs.digits) : field.digits,
            field,
            currencyField: field.currency_field,
        };
        return formatter(record.data[fieldName], formatOptions);
    }
```

Reply:
@SimonGenin 
> @h4818 Could you try
> 
> ```js
> const currencyField = options.currencyId || "currency_id";
> ```

That does not work either unfortunately :confused: 
It is the same problem for both `options.currencyField` and `options.currencyId`.
As mentioned in the PR description, the options are passed from `getFormattedValue`, and both those keys are absent and thus remain undefined.

Reply:
@FrancoisGe 
> Why not pass the currencyField in the formatOptions ? (getFormattedValue(column, record))

We should also pass `currencyId` then, no ?
This would indeed work, but it feels weird adding currency specific logic in a generic formatter ?
Which is why we thought it made more sense to fix `formatMonetary`.
But it still feels like a workaround :man_shrugging: 
We thought you might want to have a closer look at the root cause of the issue :slightly_smiling_face:

Reply:
@

> @FrancoisGe
> 
> > Why not pass the currencyField in the formatOptions ? (getFormattedValue(column, record))
> 
> We should also pass `currencyId` then, no ? This would indeed work, but it feels weird adding currency specific logic in a generic formatter ? Which is why we thought it made more sense to fix `formatMonetary`. But it still feels like a workaround 🤷‍♂️ We thought you might want to have a closer look at the root cause of the issue 🙂

```js
const formatOptions = {
            escape: false,
            data: record.data,
            isPassword: "password" in column.rawAttrs,
            digits: column.rawAttrs.digits ? JSON.parse(column.rawAttrs.digits) : field.digits,
            field: record.fields[fieldName],
        };
```
Actually, we already pass a specific set of values to the logic of a formatter (isPassword, digits)

Reply:
> @FrancoisGe
> 
> > Why not pass the currencyField in the formatOptions ? (getFormattedValue(column, record))
> 
> We should also pass `currencyId` then, no ? This would indeed work, but it feels weird adding currency specific logic in a generic formatter ? Which is why we thought it made more sense to fix `formatMonetary`. But it still feels like a workaround 🤷‍♂️ We thought you might want to have a closer look at the root cause of the issue 🙂

Actually, we already pass a specific set of values to the logic of a formatter.

```js
const formatOptions = {
            escape: false,
            data: record.data,
            isPassword: "password" in column.rawAttrs,
            digits: column.rawAttrs.digits ? JSON.parse(column.rawAttrs.digits) : field.digits,
            field: record.fields[fieldName],
        };
```

Reply:
> FYI @SimonGenin @FrancoisGe
> 
> While further investigating the issue, I realized that the fix we are suggesting here was actually the initial behavior before it was removed in this commit:
> 
> [48ef812#diff-5473186dee1246e7f729ecbd33c061aa145fda186906182825eb3f8f5ffd4f1fL283](https://github.com/odoo/odoo/commit/48ef812a635f70571b395f82ffdb2969ce99da9e#diff-5473186dee1246e7f729ecbd33c061aa145fda186906182825eb3f8f5ffd4f1fL283)

The commit is really big. Could you be a bit more specific?

Reply:
> The commit is really big. Could you be a bit more specific?

Indeed, sorry, I thought I added the line number in the link, here a screenshot:
![2022-09-28_14-57](https://user-images.githubusercontent.com/63167666/192784580-fa469787-5578-4ca2-b741-1b1ba572a6bf.png)